xml_loader/shift_generation.py

```diff
@@ -0,0 +1,261 @@
+from gurobipy import *
+from pathlib import Path
+from xml_loader.xml_loader import *
+import xml.etree.ElementTree as ET
+
+
+def get_time_steps(root):
+    demands = get_demand_definitions(root)
+    time_step_length = 100
+    for demand in demands:
+        for i in range(len(demand.end)):
+            if(demand.end[i] - int(demand.end[i]) > 0):
+                if(demand.end[i] - int(demand.end[i]) == 0.5 and time_step_length != 0.5):
+                    time_step_length = 0.5
+                elif(demand.end[i] - int(demand.end[i]) in [0.25,0.75] and time_step_length != 0.25):
+                    time_step_length = 0.25
+                elif(demand.end[i] - int(demand.end[i]) < 0.25):
+                    time_step_length = 100/60
+                    break
+            if(demand.start[i] - int(demand.start[i]) > 0):
+                if(demand.start[i] - int(demand.start[i]) == 0.5 and time_step_length < 0.5):
+                    time_step_length = 0.5
+                elif(demand.start[i] - int(demand.start[i]) == [0.25,0.75] and time_step_length < 0.25):
+                    time_step_length = 0.25
+                elif(demand.start[i] - int(demand.start[i]) < 0.25):
+                    time_step_length = 100/60
+                    break
+            if(time_step_length == 100 and (demand.end[i] - int(demand.end[i])) == 0 and (demand.start[i] - int(demand.start[i])) == 0):
+                time_step_length = 1
+    return time_step_length
+
+
+def get_time_periods(root):
+    time_periods = []
+    i = 0
+    time_step = get_time_steps(root)
+    demands = get_days_with_demand(root)
+    time_periods_in_week = tupledict()
+    week = 0
+    time_periods_in_week[week] = tuplelist()
+    for dem in demands:
+        for i in range(len(demands[dem].start)):
+            time = demands[dem].start[i] + 24*(dem)
+            end = demands[dem].end[i] + 24*dem
+            #Håndterer special cases hvor demand end er mindre enn demand start
+            if(end <= time):
+                end += 24
+            while(time < end):
+                if(time > (week+1)*24*7):
+                    week += 1
+                    time_periods_in_week[week] = []
+                if(time not in time_periods):
+                    time_periods.append(time)
+                    time_periods_in_week[week].append(time)
+                time += time_step
+    return [time_periods, time_periods_in_week]
```

Comment:
`[]` har ingen effekt, så du kan fjerne dem ;)

Reply:
Er du 100% sikker på at det ikke har noe å si for resten? Slik jeg ser det returnerer jeg nå en array istedenfor en tuple

Reply:
Du har helt rett! Regnte ikke med at du trengte en liste, men hvis du gjør det så er dette måten å gjøre det på ja :)